src/gui/options_imp.cpp

```diff
@@ -403,6 +406,18 @@ void options_imp::saveOptions()
     Preferences::setTorrentFileAssoc(checkAssociateTorrents->isChecked());
     Preferences::setMagnetLinkAssoc(checkAssociateMagnetLinks->isChecked());
 #endif
+#ifdef Q_OS_MAC
+    if (checkAssociateTorrents->isChecked()) {
+        Preferences::setTorrentFileAssoc();
+        checkAssociateTorrents->setChecked(Preferences::isTorrentFileAssocSet());
+        checkAssociateTorrents->setEnabled(!checkAssociateTorrents->isChecked());
```

Comment:
I've recorded a video how does this work, sorry for external link, the file was bigger than 10 MB https://youtu.be/IRwwSYilV7A